### PR TITLE
Add resizable/undecorated lwjgl config options.

### PR DESCRIPTION
**Pull request by bgroenks96**
Additional changes:
- Add 'forceDisplayModeChange' flag in LwjglGraphics to force
  'setWindowedMode' to reconfigure the Display after a call to
  'setUndecorated'

### CODE REVIEW

**Comment by 3xp0n3nt** [comment]
Is there a reason for this not being merged? `Gdx.graphics.setUndecorated` doesn't work as-is, and this is the fix.

**Comment by bgroenks96** [comment]
Resync 4c6397c fixes a possible bug caused by `forceDisplayModeChange` not being unset.

File: backends/gdx-backend-lwjgl/src/com/badlogic/gdx/backends/lwjgl/LwjglGraphics.java [hunk 1]
```diff
@@ -79,7 +80,7 @@
 	}
 
 	LwjglGraphics (Canvas canvas, LwjglApplicationConfiguration config) {
-		this.config = config;
+		this(config);
```
Comment (intrigus) [inline backends/gdx-backend-lwjgl/src/com/badlogic/gdx/backends/lwjgl/LwjglGraphics.java]: Why did you change this to a constructor call?

**Comment by intrigus** [comment]
I'm in favor of having a new method `setWindowedMode (int width, int height, boolean force)` instead of the `forceDisplayModeChange` flag.

The current implementation makes it impossible to "un-force" display mode changes.

File: backends/gdx-backend-lwjgl/src/com/badlogic/gdx/backends/lwjgl/LwjglGraphics.java (hunk 1 above)
Comment (bgroenks96) [inline backends/gdx-backend-lwjgl/src/com/badlogic/gdx/backends/lwjgl/LwjglGraphics.java]: I think I originally changed something in the constructor, then got rid of the change. But I guess I left the `this` call in. It looks like it's functionally the same, though? Seems more directionally correct to re-use constructor code rather than duplicate it, right?

**Comment by bgroenks96** [comment]
> The current implementation makes it impossible to "un-force" display mode changes.

I don't get what you mean. Currently, if you call `setUndecorated`, the display mode needs to be reset in order for LWJGL to update the window. The `forceDisplayModeChange` signals to the `setWindowedMode` method that it needs to do this even if the display size hasn't changed. Why would you ever need to "un-force" that change?

I'm reluctant to make another change to the Graphics SPI. We could change it, however, such that `setWindowedMode (int width, int height, boolean force)` is a private method called by both `setUndecorated` and the public `setWindowedMode` methods. That would also remove the current necessity of calling `setWindowedMode` after calling `setUndeocrated`

**Comment by 3xp0n3nt** [comment]
> intrigus commented a day ago
>
> I'm in favor of having a new method setWindowedMode (int width, int height, boolean force) instead of the forceDisplayModeChange flag.
>
> The current implementation makes it impossible to "un-force" display mode changes.

Sorry, but this is completely unnecessary, as @bgroenks96 implied, it is an implementation detail. There is absolutely no need to obfuscate the API in such a way. If it's so important, than I'm sure you'll have no problem providing a detailed use case whereby said proposed changes would prevent you from accomplishing something? I can think of no such example.

File: backends/gdx-backend-lwjgl/src/com/badlogic/gdx/backends/lwjgl/LwjglGraphics.java (hunk 1 above)
Comment (3xp0n3nt) [inline backends/gdx-backend-lwjgl/src/com/badlogic/gdx/backends/lwjgl/LwjglGraphics.java]: @intrigus It really doesn't matter either way, seeing as there are only two field assignments. If, however, there were more occurring in the constructors, it would make much more sense to re-use them. This is a common pattern, both in this library and in general (as long as you don't abuse it by telescoping params, in which case you should consider using a builder).

**Comment by intrigus** [comment]
> > The current implementation makes it impossible to "un-force" display mode changes.
> 
> I don't get what you mean. Currently, if you call setUndecorated, the display mode needs to be reset in order for LWJGL to update the window. The `forceDisplayModeChange` signals to the `setWindowedMode` method that it needs to do this even if the display size hasn't changed. Why would you ever need to "un-force" that change?
> 
> I'm reluctant to make another change to the Graphics SPI. We could change it, however, such that `setWindowedMode (int width, int height, boolean force)` is a private method called by both `setUndecorated` and the public `setWindowedMode` methods. That would also remove the current necessity of calling `setWindowedMode` after calling `setUndeocrated`

The problem is that after you have _once_ called `setUndecorated`,  `setWindowedMode` will _always_ change the windowed mode, because `forceDisplayModeChange` is still `true` and never set to `false`.

Btw. calling `setWindowedMode` after a call to Lwjgl3's `setUndecorated` does [not do](https://github.com/libgdx/libgdx/pull/3847/files#diff-82769c2b87264103e89de2b6613febf4R374) what it should do.

File: backends/gdx-backend-lwjgl/src/com/badlogic/gdx/backends/lwjgl/LwjglGraphics.java (hunk 1 above)
Comment (intrigus) [inline backends/gdx-backend-lwjgl/src/com/badlogic/gdx/backends/lwjgl/LwjglGraphics.java]: @3xp0n3nt I know that it doesn't matter. I also know this pattern, but people normally have a reason for a change :)

**Comment by bgroenks96** [comment]
>forceDisplayModeChange is still true and never set to false.

That was supposed to have been fixed in 4c6397c by setting the flag to false in `setWindowedMode`. It looks like the change didn't show up (I'm just now noticing that). Maybe I forgot to put `-a` when I amended the commit 😰 

I'll fix that that when I am able later today and amend the commit again.

>Btw. calling setWindowedMode after a call to Lwjgl3's setUndecorated does not do what it should do.

Can you create an issue and elaborate on this? I haven't tried using the new APIs with the LWJGL3 backend. If you let me know what's wrong, I could probably add a quick fix to this PR.

**Comment by badlogic** [comment]
Please resubmit if you fixed things up.
